code/game/machinery/machinery.dm

```diff
@@ -465,6 +465,22 @@ Class Procs:
 	M.set_build_state(2)
 	M.state = 1
 
+/obj/machinery/proc/spillContents(var/destroy_chance = 0)
+	for(var/obj/I in component_parts)
+		if(prob(destroy_chance))
+			qdel(I)
+		else
+			if(istype(I, /obj/item/weapon/reagent_containers/glass/beaker) && src:reagents && src:reagents.total_volume)
+				reagents.trans_to(I, reagents.total_volume)
+			if(I.reliability != 100 && crit_fail)
+				I.crit_fail = 1
+			I.loc = src.loc
+	for(var/obj/I in src) //remove any stuff loaded, like for fridges

```

Comment:
Probably a good idea to make this `/atom/movable`, in the case of things like freezers.